update Code of Conduct to version 2

zkamvar opened the pull request:
The previous version of the Code of Conduct (CoC) contained language that was too specific to contributions to code repositories. Moreover, there was no clear definition of what the consequences for those who violated the CoC were beyond the scope of code. This version of the CoC fills the gaps a lot better.

I would like input from the community about these changes.

zkamvar committed:
update CoC to version 2

The previous version of the CoC contained language that was too specific to contributions to code repositories. Moreover, there was no clear definition of what the consequences for those who violated the CoC were beyond the scope of code. This version of the CoC fills the gaps a lot better.

zkamvar committed:
update to include slack and email

AmmerB commented on CODE_OF_CONDUCT.md:
```diff
@@ -1,27 +1,90 @@
 
-As contributors and maintainers of this project, we pledge to respect all people who 
-contribute through reporting issues, posting feature requests, updating documentation,
-submitting pull requests or patches, and other activities.
```
- Contribute to RECON training activities or deployments

zkamvar replied:
You're commenting on deleted text here. I believe this is covered in the "Scope" paragraph below: 

> This Code of Conduct applies within all RECON community spaces

AmmerB commented on CODE_OF_CONDUCT.md:
```diff
@@ -1,27 +1,90 @@
+https://www.contributor-covenant.org/faq. Translations are available at https://www.contributor-covenant.org/translations.
 
-Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by 
-opening an issue or contacting one or more of the project maintainers.
```
Such issues will discussed by the RECON advisory board and appropriate actions will be taken after careful and deliberate review of such complaints.

zkamvar replied:
This is on removed text, but is covered in the Enforcement section:

> All complaints will be reviewed and investigated promptly and fairly.
>
> All community leaders are obligated to respect the privacy and security of the reporter of any incident.

AmmerB commented on CODE_OF_CONDUCT.md:
```diff
@@ -1,27 +1,90 @@
+
+**Community Impact**: Demonstrating a pattern of violation of community standards, including sustained inappropriate behavior,  harassment of an individual, or aggression toward or disparagement of classes of individuals.
+
+**Consequence**: A permanent ban from any sort of public interaction within the RECON community.
```
If required legal action can be taken, or such action can be reported to the appropriate authorities

zkamvar replied:
I'm not sure how this sentence works. How about this:

```suggestion
**Consequence**: A permanent ban from any sort of public interaction within the RECON community. 
Any action requiring legal intervention will be reported to the appropriate authorities.
```

thibautjombart replied:
I think the comment on legal intervention should be made before detailing the different grades of response, as it applies at least in theory to all of them.

scolombe commented on CODE_OF_CONDUCT.md:
```diff
@@ -1,27 +1,90 @@
-Examples of unacceptable behavior by participants include the use of sexual language or
-imagery, derogatory comments or personal attacks, trolling, public or private harassment,
-insults, or other unprofessional conduct.
+We as members, contributors, and leaders pledge to make participation in the RECON community a harassment-free experience for everyone, regardless of age, body size, visible or invisible disability, ethnicity, sex characteristics, gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, or sexual identity and orientat ...
```
```suggestion
We as members, contributors, and leaders pledge to make participation in the RECON community a harassment-free experience for everyone, regardless of age, body size, visible or invisible disability, ethnicity, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, actual or perceived sexual identity and orientation, gender identity and expression, or sex characteristics. 
```

scolombe commented on CODE_OF_CONDUCT.md:
```diff
@@ -1,27 +1,90 @@
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be reported to the community leaders responsible for enforcement by emailing [repidemicsconsortium@gmail.com](mailto:repidemicsconsortium@gmail.com?subject=CoCEnforcement) or contacting one of the [Executive Board](https://www.repidemicsconsortium.org/people#executive-board) members via the [RECON slack channel](https://recon ...
+
```
Use a separate email address and to be discussed whether the executive board is the best contact to receive first hand complaints.

cwhittaker1000 committed:
Added in Soledad's additions to protected characteristics, updated reporting guidelines to highlight new google form and email, and added attribution to rOpenSci for inspiration for the reporting guidelines.

cwhittaker1000 commented:
Have just pushed a new commit providing updates to the Code of Conduct. These updates are:
1. Incorporating in @scolombe's additions and extensions to the protected characteristics section.
2. Incorporating in detail about the new reporting guidelines being put in place (specifically, the new google form, the new email with automatic onwards forwarding to the CoC committee etc).
3. Attribution to rOpenSci, recognising their current reporting guidelines as the inspirations for the system we have put in place here.
Aim is to get feedback on this from the board and pass at the next RECON board meeting (early January).

cwhittaker1000 committed:
alterations to reporting guidelines to incorporate the need for the executive board to vote on CoC Committee Recommendations for Action

cwhittaker1000 committed:
Update code of conduct following today's board meeting

Added in the following after today's board meeting:
1. Official documentation stating that trainers and participants will be expected to sign up to the code of conduct upon registering for a RECON workshop or training event
2. Noting that in addition to broader punishments (generalised bans, revoking of membership etc) that the CoC Committee reserves the right to implement targeted sanctions surrounding e.g. banning from teaching or participating in RECON events.

cwhittaker1000 committed:
new google forms link

cwhittaker1000 committed:
grammar error fix

cwhittaker1000 committed:
update COC
